### PR TITLE
Add documentation for new templates system

### PR DESCRIPTION
**Pull request by rschroll**
This is to be applied on top of #1601.

### CODE REVIEW

**Comment by rschroll** [comment]
I tried to build the docs locally, but ran into problems with a missing theme.  I think everything is reasonable, but do double check me.

File: docs/source/reference/templates.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Templates
+
+The pages of the JupyterHub application are generated from [Jinja](http://jinja.pocoo.org/) templates.  These allow the header, for example, to be defined once and incorporated into all pages.  By providing your own templates, you can have complete control over JupyterHub's appearance.
+
+## Custom Templates
+
+JupyterHub will look for custom templates in all of the paths in the `JupyterHub.template_paths` configuration option, falling back on the [default templates](https://github.com/jupyterhub/jupyterhub/tree/master/share/jupyter/hub/templates) if no custom template with that name is found.  This means you can override as many or as few templates as you desire.
```
Comment (minrk) [inline docs/source/reference/templates.md]: Can you add a little note that template_paths is new in 0.9?

Comment (minrk) [reply docs/source/reference/templates.md]: Oops, rather than `{% extends templates/... %}` support is new. This config isn't the new part.

**Comment by rschroll** [comment]
Try this on for size.

**Comment by minrk** [comment]
Perfect, thanks!

**Pull request merged**